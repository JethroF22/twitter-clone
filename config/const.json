{
  "actionStatusMessages": {
    "FAILED_MESSAGE": "Failed",
    "IN_PROGRESS_MESSAGE": "In Progress",
    "SUCCESS_MESSAGE": "Successful"
  },
  "actionTypes": {
    "auth": {
      "SET_USER_DETAILS": "SET_USER_DETAILS",
      "SET_TOKEN": "SET_TOKEN"
    },
    "error": {
      "SET_ERROR_MESSAGE": "SET_ERROR_MESSAGE"
    },
    "profile": {
      "SET_USER_PROFILE": "SET_USER_PROFILE",
      "VIEW_PROFILE": "VIEW_PROFILE"
    },
    "status": {
      "SET_ACTION_STATUS": "SET_ACTION_STATUS"
    },
    "tweet": {
      "SET_USER_TWEETS": "SET_USER_TWEETS",
      "UPDATE_USER_TWEETS": "UPDATE_USER_TWEETS"
    }
  },
  "errorTypes": {
    "AUTHENTICATION_ERROR": "Authentication Error",
    "AUTHORISATION_ERROR": "Authorisation Error",
    "DB_ERROR": "DB Error",
    "INVALID_REQUEST": "Invalid Request"
  },
  "errorMessages": {
    "CANT_BE_FOLLOWED": "Users cannot follow themselves",
    "DUPLICATE_EMAIL": "Duplicate email",
    "DUPLICATE_HANDLE": "Duplicate handle",
    "HAS_BEEN_LIKED": "User has already liked this tweet",
    "HAS_NOT_BEEN_LIKED": "User has yet to like this tweet",
    "HAS_BEEN_RETWEETED": "Cannot retweet the same tweet more than once",
    "HAS_BEEN_FOLLOWED": "Already following user",
    "HAS_NOT_BEEN_FOLLOWED": "User is not currently being followed",
    "INVALID_CREDENTIALS": "Invalid email/password combination",
    "TWEET_NOT_FOUND": "Tweet not found",
    "TWEETS_NOT_FOUND": "User has yet to tweet anything",
    "UNKNOWN_ERROR": "Unknown DB error",
    "UNAUTHORISED": "Unauthorised",
    "USER_NOT_FOUND": "User not found",
    "VALIDATION_ERROR": "Validation Error"
  }
}
