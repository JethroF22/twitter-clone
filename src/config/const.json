{
  "actionStatusMessages": {
    "SUCCESS_MESSAGE": "Successful",
    "IN_PROGRESS_MESSAGE": "In Progress",
    "FAILED_MESSAGE": "Failed"
  },
  "actionTypes": {
    "auth": {
      "SET_USER_DETAILS": "SET_USER_DETAILS",
      "SET_TOKEN": "SET_TOKEN"
    },
    "status": {
      "SET_ACTION_STATUS": "SET_ACTION_STATUS"
    },
    "error": {
      "SET_ERROR_MESSAGE": "SET_ERROR_MESSAGE"
    }
  }
}
