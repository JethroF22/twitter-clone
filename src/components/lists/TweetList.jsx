import React, { Component } from 'react';
import { connect } from 'react-redux';

import Tweet from './list-items/Tweet';
import { fetchTweets } from '../../actions/tweet';
import { actionStatusMessages } from '../../../config/const.json';

const { SUCCESS_MESSAGE } = actionStatusMessages;

export class TweetList extends Component {
  componentDidMount() {
    this.props.fetchTweets(this.props.id);
  }

  render() {
    return (
      <div>
        {this.props.actionStatus === SUCCESS_MESSAGE ? (
          this.props.tweets.length > 0 ? (
            this.props.tweets.map(tweet => (
              <Tweet tweet={tweet} key={Math.random() * 99999} />
            ))
          ) : (
            <p>No tweets to display</p>
          )
        ) : (
          <p>Loading ...</p>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  tweets: state.tweet.tweets,
  actionStatus: state.status.actionStatus
});

const mapDispatchToProps = dispatch => ({
  fetchTweets: id => dispatch(fetchTweets(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TweetList);
