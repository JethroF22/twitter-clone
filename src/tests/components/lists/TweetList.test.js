import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';

import { TweetList } from '../../../components/lists/TweetList';
import { tweets as originalTweets, users } from '../../../../config/seed';
import { actionStatusMessages } from '../../../../config/const.json';
const {
  SUCCESS_MESSAGE,
  IN_PROGRESS_MESSAGE,
  FAILED_MESSAGE
} = actionStatusMessages;

let wrapper, fetchTweetsSpy, id, tweets;

describe('TweetList', () => {
  beforeEach(() => {
    fetchTweetsSpy = jest.fn().mockResolvedValueOnce(null);
    id = users[0]._id;
    tweets = originalTweets.map(tweet => tweet.body);
    wrapper = shallow(
      <TweetList
        fetchTweets={fetchTweetsSpy}
        tweets={tweets}
        actionStatus={SUCCESS_MESSAGE}
        id={id}
      />
    );
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should dispatch action on mount', () => {
    expect(fetchTweetsSpy).toHaveBeenLastCalledWith(id);
  });

  describe('renders the correct content', () => {
    test('should display a message if there are no tweets to display', () => {
      wrapper = shallow(
        <TweetList
          fetchTweets={fetchTweetsSpy}
          tweets={[]}
          actionStatus={SUCCESS_MESSAGE}
          id={id}
        />
      );
      expect(wrapper.contains(<p>No tweets to display</p>)).toBeTruthy();
    });

    test('should display a message if the action is in progress', () => {
      wrapper = shallow(
        <TweetList
          fetchTweets={fetchTweetsSpy}
          tweets={[]}
          actionStatus={IN_PROGRESS_MESSAGE}
          id={id}
        />
      );
      expect(wrapper.contains(<p>Loading ...</p>)).toBeTruthy();
    });
  });
});
