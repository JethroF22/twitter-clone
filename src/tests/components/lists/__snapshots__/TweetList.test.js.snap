// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetList should render correctly 1`] = `
<div>
  <Tweet
    key="52054.75047841195"
    tweet="This is a test"
  />
  <Tweet
    key="41946.16570024396"
    tweet="Tres tristes tigres tragaban trigo en tres tristes trastos"
  />
</div>
`;
